src/components/app.tsx

```diff
@@ -774,38 +804,155 @@ export class App extends React.Component<AppProps, AppState> {
+        return (
+            <div className="sensor-controls-holder">
+                {!wiredConnected && !wirelessConnected ?
+                    <div className="connect-message-holder">
+                        <div className="connect-message">{this.messages["connection_message"]}</div>
+                        <div className="connect-sub-message">{this.messages["connection_sub_message"]}</div>
+                    </div>
+                    : null
+                }
+                <div className="sensor-buttons">
+                    {this.renderConnectionButtons()}
+                </div>
+                {this.renderGraphTopPanels()}
+            </div>
+        );
+    }
+
+    renderConnectionButtons() {
+        const { sensorManager } = this.state;
+        const wirelessConnected = sensorManager && sensorManager.isWirelessDevice();
+        const wiredConnected = sensorManager && !sensorManager.isWirelessDevice();
+        if (!this.props.sensorManager) {
+            return (
+                <div>
+                    { !wiredConnected && !wirelessConnected ?
+                    <div>
+                        <button className="connect-to-device-button smart-focus-highlight disable-focus-highlight"
+                                onClick={this.handleWirelessClick}
+                                disabled={wiredConnected || this.state.collecting}>
+                            Wireless Sensor
+                        </button>
+                        <button className="connect-to-device-button smart-focus-highlight disable-focus-highlight"
+                                onClick={this.handleWiredClick}
+                                disabled={wirelessConnected || this.state.collecting}>
+                            Wired Sensor
+                        </button>
+                    </div>
+                    : null }
+                </div>
+            );
         } else {
-          return  <Button className="connect-to-device-button" onClick={this.connectToDevice} >
-                    Connect to Device
-                  </Button>;
+            // Check if this sensorManger supports device connection
+            if (isConnectableSensorManager(sensorManager)) {
+                if (sensorManager.deviceConnected) {
+                return  <Button className="connect-to-device-button" onClick={this.disconnectFromDevice} >
+                            Disconnect Device
+                        </Button>;
+                } else {
+                return  <Button className="connect-to-device-button" onClick={this.connectToDevice} >
+                            Connect Device
+                        </Button>;
+                }
+            } else {
+                return null;
+            }
         }
+    }
 
-      } else {
-        return null;
-      }
+    renderGraphTopPanels() {
+        const { sensorManager, sensorSlots } = this.state;
+        const connected = sensorManager !== null;
```

Comment:
```suggestion
        const connected = sensorManager != null;
```
in case `sensorManager === undefined`.

Reply:
Fixed.